test/integration/vaults/pdg.integration.ts

```diff
@@ -87,47 +96,173 @@ describe("Integration: Predeposit Guarantee core functionality", () => {
-  //    Method called: Dashboard.proveUnknownValidatorsToPDG(witness).
-  // 6. The Oracle report confirms the validator’s balance (1 ETH). The stVault’s total value is then increased by 1 ETH accordingly.
-  // 7. The Node Operator deposits the remaining 99 ETH from the vault balance to the validator through the PDG.
-  //    Method called: PredepositGuarantee.depositToBeaconChain(stakingVault, deposits).
-
-  // Works with vaults deposit pauses
-  // Steps 1-7 are the same as in the full cycle trustless path
-  // 8. The stVault's owner pauses the vault's deposits.
-  //    Method called: Dashboard.pauseBeaconChainDeposits().
-  // 9. The Node Operator tries to deposit the remaining 99 ETH from the vault balance to the validator through the PDG.
-  //    Method called: PredepositGuarantee.depositToBeaconChain(stakingVault, deposits).
-  //    This reverts with the "BeaconChainDepositsPaused" error.
-  // 10. The stVault's owner resumes the vault's deposits.
-  //    Method called: Dashboard.resumeBeaconChainDeposits().
-  // 11. The Node Operator deposits the remaining 99 ETH from the vault balance to the validator through the PDG.
-  //    Method called: PredepositGuarantee.depositToBeaconChain(stakingVault, deposits).
+  it.skip("PDG shortcut", async () => {
+    const { predepositGuarantee } = ctx.contracts;
+
+    // 1. The stVault's owner supplies 100 ETH to the vault.
+    await expect(dashboard.connect(roles.funder).fund({ value: ether("100") }))
+      .to.emit(stakingVault, "Funded")
+      .withArgs(dashboard, ether("100"));
+
+    // 2. The Node Operator generates validator keys and deposit data.
+    const withdrawalCredentials = await stakingVault.withdrawalCredentials();
+    const validator = generateValidator(withdrawalCredentials);
+
+    // 3. The Node Operator shares the deposit data with the stVault's owner.
+    // (This is a conceptual step, no actual code needed)
+
+    const predepositData = await generatePredepositData(predepositGuarantee, dashboard, roles, nodeOperator, validator);
+
+    //Tracing.enable();
+
+    await dashboard.connect(owner).grantRole(await dashboard.FUND_ROLE(), proxy);
+
+    // todo: I suspect that this report is issued with some data that causes failure in thext method
+    await reportVaultDataWithProof(stakingVault);
+
+    // 4. The stVault's owner deposits 1 ETH from the vault balance directly to the validator, bypassing the PDG.
+    //    Method called: Dashboard.unguaranteedDepositToBeaconChain(deposits).
+    //    4.1. As a result, the stVault's total value is temporarily reduced by 1 ETH until the next oracle report delivered containing the appeared validator's balance.
+    // todo: this step fails, BUT this is the point of the test!
+    await expect(
+      dashboard
+        .connect(roles.unguaranteedBeaconChainDepositor)
+        .unguaranteedDepositToBeaconChain([predepositData.deposit]),
```

Comment:
Here we have a loop with funds transferring inside the `unguaranteedDepositToBeaconChain` function! 
This test actually found the issue! Great work! 👏 

Let's remove .skip, because it does a correct check! And then we can merge it.

Reply:
done!